bin/proving-service/src/proxy/worker.rs

```diff
@@ -309,20 +310,26 @@ async fn create_status_client(
 
 /// Returns whether the worker version is valid.
 ///
-/// The version is valid if it is a semantic version and is greater than or equal to the
-/// current version. We dont check the patch version.
-/// Returns false if either version string is malformed.
-fn is_valid_version(current_version: &str, received_version: &str) -> bool {
-    // Dont check the patch version.
-    let current_version_parts: Vec<&str> = current_version.split('.').collect();
-    let version_parts: Vec<&str> = received_version.split('.').collect();
-
-    // Check if both versions have at least major and minor components
-    if current_version_parts.len() < 2 || version_parts.len() < 2 {
-        return false;
-    }
-
-    version_parts[0] == current_version_parts[0] && version_parts[1] == current_version_parts[1]
+/// The version is valid if it is a semantic version and the major and minor versions match
+/// the current version. The patch version is ignored.
+/// Returns false if either version does not match the current version.
+/// Returns an error if the version is malformed.
+fn is_valid_version(current_version: &str, received_version: &str) -> Result<bool, String> {
```

Comment:
I would re-write the comment as:
```rust
/// Returns true if the received_version has major and minor versions match the major and minor
/// versions of the current_version. Returns false otherwise.
///
/// # Errors
/// Returns an error if either of the versions is malformed.
```
Also, `current_version` and `received_version` naming is a bit confusing. I'd probably just name them `version_req` and `version`.

Lastly, we should probably pre-parse the `current_version` into `VersionReq` (maybe using something like `LazyLock` struct?).